numpy/core/tests/test_arrayprint.py

```diff
@@ -259,7 +259,7 @@ def _format_function(x):
                 '[abcabc defdef]')
 
 
-    def test_structure_format(self):
+    def test_structure_format_mixed(self):
```

Comment:
I split this test into 3, since the other test functions were not related to this one. I hit this by accident when I was trying out a different version of the code and printing was failing.